### PR TITLE
[core][dependencies] test pre-release dependencies as part of CI

### PR DESCRIPTION
**Pull request by scv119**
## Why are these changes needed?

Install pre release version of unbounded dependencies and run smoke test

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ci/ci.sh [hunk 1]
```diff
@@ -755,10 +755,7 @@ build() {
   fi
 }
 
-test_minimal() {
```
Comment (cadedaniel) [inline ci/ci.sh:758]: How was this called before? 
```
$ ag 'test_minimal'
python/ray/train/BUILD
242:    name = "test_minimal",
244:    srcs = ["tests/test_minimal.py"],

ci/ci.sh
758:test_minimal() {
```

File: ci/env/install-core-prerelease-dependencies.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+# unbounded dependencies in setup.py for ray core
+for dependency in attrs grpcio jsonschema aiosignal frozenlist requests
```
Comment (cadedaniel) [inline ci/env/install-core-prerelease-dependencies.sh:4]: good idea! what should we do when this fails? fine to defer until that happens but it seems to me we need a mechanism to ignore broken releases.

**Review by cadedaniel — APPROVED** [review]

File: ci/ci.sh (hunk 1 above)
Comment (cadedaniel) [reply ci/ci.sh:758]: ah some buildkite magic invokes it
```
BUILDKITE_COMMAND="echo '--- :arrow_forward: cleanup() { if [ \"28623\" = \"false\" ]; then ./ci/build/upload_build_info.sh; fi }; trap cleanup EXIT'
cleanup() { if [ \"28623\" = \"false\" ]; then ./ci/build/upload_build_info.sh; fi }; trap cleanup EXIT
echo '--- :arrow_forward: ./ci/ci.sh test_minimal 3.6'
```

File: ci/env/install-core-prerelease-dependencies.sh (hunk 2 above)
Comment (scv119) [reply ci/env/install-core-prerelease-dependencies.sh:4]: yeah that's a good question.

**Comment by scv119** [comment]
hmm looks the new step doesn't run in the ci https://buildkite.com/ray-project/oss-ci-build-pr/builds/257#0183677c-d335-4346-b335-acef1f8a37b2

@simon-mo do you know what that's the case?

**Comment by simon-mo** [comment]
@krfricke

**Pull request merged**